### PR TITLE
Allow RPC bind address to be reconfigured

### PR DESCRIPTION
**Pull request by lemmih**
**Summary of changes**
Changes introduced in this pull request:
- Allow the RPC address to be changed to `0.0.0.0`.
- It can be useful when accessing forest from a different container/machine.
- Security is still handled via access tokens.

**Reference issue to close (if applicable)**

Closes 

**Other information and links**

### CODE REVIEW

**Review by tyshko5 — APPROVED** [review]

File: forest/src/cli/config.rs [hunk 1]
```diff
@@ -31,6 +31,8 @@ pub struct Config {
     pub encrypt_keystore: bool,
     /// Metrics bind, e.g. 127.0.0.1:6116
     pub metrics_address: SocketAddr,
+    /// RPC bind, e.g. 127.0.0.1
```
Comment (LesnyRumcajs) [inline forest/src/cli/config.rs:34]: How about we use the same approach as in `metrics_address` and use `SocketAddr`? This way, on top of some uniformity, we'll have both the address and the port in one param.

Comment (lemmih) [reply forest/src/cli/config.rs:34]: Good idea.

Comment (LesnyRumcajs) [reply forest/src/cli/config.rs]: Thanks. Yeah, it'd be great to specify the port purpose in the name.

File: forest/src/cli/config.rs [hunk 2]
```diff
@@ -31,12 +29,17 @@ pub struct Config {
...
+    /// RPC bind, e.g. 127.0.0.1:1234
+    pub rpc_address: SocketAddr,
     pub rocks_db: forest_db::rocks_config::RocksDbConfig,
     pub network: Libp2pConfig,
     pub sync: SyncConfig,
     pub chain: Arc<ChainConfig>,
 }
 
+// 127.0.0.1
+const LOCALHOST: IpAddr = IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1));
```
Comment (LesnyRumcajs) [inline forest/src/cli/config.rs:41]: Do we need it? There's a constant `std::net::Ipv4Addr::LOCALHOST` already there for grabbing.

**Review by LesnyRumcajs — APPROVED** [review]

File: forest/src/cli/config.rs (hunk 2 above)
Comment (lemmih) [reply forest/src/cli/config.rs]: I learn something new every day. :)

File: forest/src/cli/auth_cmd.rs [hunk 3]
```diff
@@ -57,7 +57,7 @@ impl AuthCommands {
                     Ok(token) => {
                         let mut addr = Multiaddr::empty();
                         addr.push(Protocol::Ip4(DEFAULT_HOST.parse().unwrap()));
```
Comment (LesnyRumcajs) [inline forest/src/cli/auth_cmd.rs:59]: in general, shouldn't this be overridden as well?

Comment (lemmih) [reply forest/src/cli/auth_cmd.rs]: It should. Fixed.

**Pull request merged**